src/v/kafka/server/handlers/fetch.h

```diff
@@ -361,7 +361,11 @@ struct fetch_plan {
     explicit fetch_plan(
       size_t shards,
       op_context::latency_point start_time = op_context::latency_clock::now())
-      : fetches_per_shard(shards, shard_fetch(start_time)) {}
+      : fetches_per_shard(shards, shard_fetch(start_time)) {
+        for (size_t i = 0; i < fetches_per_shard.size(); i++) {
+            fetches_per_shard[i].shard = i;
```

Comment:
It's better (cleaner) to update the constructor to take this shard parameter, vs half-constructing the using object in the vector constructor then finishing it off here.

It is annoying that it's hard to do this right in the vector constructor. I guess the "official" way to do this efficiently in the consructor is something like transforming an iota_view into a `shard_fetch` object, then converting "using" that range to initalize the vector. I think that requires an intermediate function to pass the range into, gah.

Reply:
(not suggesting any change here)